pkgs/applications/office/tks/default.nix

```diff
@@ -0,0 +1,94 @@
+{ dpkg, fetchurl, makeWrapper, perl, stdenv }:
+
+let
+  perlEnv = perl.withPackages (p: with p; [
+    CarpClan
+    ClassLoad
+    ClassLoadXS
+    ClassMethodMaker
+    ConfigIniFiles
+    DataOptList
+    DateCalc
+    DBDSQLite
+    DBI
+    DevelGlobalDestruction
+    DevelOverloadInfo
+    EncodeLocale
+    EvalClosure
+    ExporterTiny
+    FileSlurp
+    HTMLForm
+    HTMLParser
+    HTMLTagset
+    HTTPCookies
+    HTTPDate
+    HTTPMessage
+    IPCSystemSimple
+    JSON
+    JSONXS
+    ListMoreUtils
+    LWP
+    LWPProtocolHttps
+    ModernPerl
+    ModuleImplementation
+    ModuleRuntime
+    Moose
+    MROCompat
+    PackageDeprecationManager
+    PackageStash
+    ParamsUtil
+    SubExporter
+    SubExporterProgressive
+    SubIdentify
+    SubInstall
+    SubName
+    TermProgressBar
+    TermReadKey
+    TermReadLineGnu
+    TimeDate
+    TryTiny
+    URI
+    WWWMechanize
+    XMLLibXML
+    XMLSAXBase
+  ]);
+in
+
+stdenv.mkDerivation rec {
+  pname = "tks";
+  version = "1.0.32";
+
+  src = fetchurl {
+    url = "https://debian.catalyst.net.nz/catalyst/dists/stable/catalyst/binary-amd64/tks_${version}_all.deb";
+    sha256 = "b4e288fd1068adbe0cbef12e0a916df20a785d90f7773e862b3c9f3b9528afb6";
+  };
+
+  nativeBuildInputs = [ dpkg makeWrapper ];
+  buildInputs = [ perlEnv ];
+
+  unpackPhase = "dpkg-deb -x ${src} .";
+
+  preConfigure = ''
+    patchShebangs usr/bin/*
+  '';
+
+  installPhase = ''
+    runHook preInstall
```

Comment:
`runHook` are not needed.
They are for functions like `buildPerlModule` which provide templates for derivations which will be customised further.

Reply:
That not quite true. The pre/post hooks are run by the standard builder, so if somebody wants to easily override, this is a good place to do so and for that reason they should be left in.

Reply:
The overrider is in position to override `installPhase`

Reply:
But then the user has to reimplement the `installPhase` as well.
Having these runHook calls is useful.

Reply:
No, it does not have to reimplement, it has to call the original.
```
xxx.overrideDerivation(old: {
  installPhase = ''
    something before
    ${old.installPhase or ""}
    something after
  '';
};
```

overriding `preInstall` does not free the user up from calling the original:

Is 
```
xxx.overrideDerivation(old: {
  preInstall = ''
    something before
    ${old.preInstall or ""}
    something after
  '';
};
```
somehow better?
Does it worth extra (2*number_of_phases) code lines in every `mkDerivation`?